src/layer.rs

```diff
@@ -63,6 +63,15 @@ impl LayerSet {
         Ok(LayerSet { layers })
     }
 
+    /// Create a new `LayerSet`.
+    ///
+    /// Will panic if `layers` is empty.
+    pub fn new(mut layers: Vec<Layer>) -> Self {
+        assert!(!layers.is_empty());
```

Comment:
This irks me, but I also find Result here less ergonomic :(

Reply:
This isn't a method we would really expect people to use except in very special cases, so I'm okay with expecting a higher degree of caution.